src/SingleLineTextEdit.h

```diff
@@ -0,0 +1,44 @@
+#ifndef SINGLELINETEXTEDIT_H
+#define SINGLELINETEXTEDIT_H
+
+/***************************************************************************
+ *   This program is free software; you can redistribute it and/or modify  *
```

Comment:
this is an uncopyrighted work, I do not wish for attribution

Reply:
We need a copyright line in order to apply the GPL to the software - if it does not have a copyright then the software cannot be protected by the GPL's "Copyleft" - http://oitofelix.github.io/gnu-philosophy/standalone/pdf/what-is-copyleft.pdf ...

Reply:
> The simplest way to make a program free software is to put it in the public domain,
uncopyrighted. 

Seems like this is exactly what Zooka is doing, meaning that we can then apply the GPL licence to it, and that'll be ok?

Reply:
Take a look at https://choosealicense.com/no-permission/ and note that without any other expression of ownership:
> When you make a creative work (which includes code), the work is under exclusive copyright by default. Unless you include a license that specifies otherwise, nobody else can copy, distribute, or modify your work without being at risk of take-downs, shake-downs, or litigation. Once the work has other contributors (each a copyright holder), “nobody” starts including you.

Looking through https://choosealicense.com/ suggests that if you really want to go down that route you might want to consider the **[Unlicence](https://choosealicense.com/licenses/#unlicense)** - also bear in mind that if you do that, I or another developer (without evil intent :crossed_fingers:) could legitimately subsequently apply the GPL 2+ licence to it anyhow...